code/modules/mob/living/carbon/xenomorph/castes/king/abilities_king.dm

```diff
@@ -32,104 +32,272 @@
+	var/timer_ref
 
-/datum/action/xeno_action/activable/gravity_crush/on_cooldown_finish()
-	to_chat(owner, span_warning("Our psychic aura restores itself. We are ready to gravity crush again."))
+/datum/action/xeno_action/zero_form_beam/Destroy()
+	QDEL_NULL(sound_loop)
 	return ..()
 
-/datum/action/xeno_action/activable/gravity_crush/can_use_ability(atom/A, silent, override_flags)
+/datum/action/xeno_action/zero_form_beam/New(Target)
+	. = ..()
+	sound_loop = new
+
+/obj/effect/ebeam/zeroform/Initialize()
+	. = ..()
+	alpha = 0
+	animate(src, alpha = 255, time = ZEROFORM_CHARGE_TIME)
+
+/datum/action/xeno_action/zero_form_beam/can_use_action(silent, override_flags)
 	. = ..()
 	if(!.)
 		return
-	if(!line_of_sight(owner, A, king_crush_dist))
+	if(SSmonitor.gamestate == SHUTTERS_CLOSED && is_ground_level(owner.z))
 		if(!silent)
-			to_chat(owner, span_warning("We must get closer to crush, our mind cannot reach this far."))
+			owner.balloon_alert("too early")
 		return FALSE
+	if(LAZYACCESS(owner.do_actions, src))
+		if(!silent)
+			owner.balloon_alert(owner, "already busy")
+		return FALSE
+
+/datum/action/xeno_action/zero_form_beam/action_activate()
+	if(timer_ref)
+		stop_beaming()
+		return
 
-/datum/action/xeno_action/activable/gravity_crush/use_ability(atom/A)
-	owner.face_atom(A) //Face towards the target so we don't look silly
-	var/list/turfs = RANGE_TURFS(1, A)
-	playsound(A, 'sound/effects/bomb_fall.ogg', 50, FALSE)
-	apply_filters(turfs)
-	if(!do_after(owner, WINDUP_GRAV, FALSE, owner, BUSY_ICON_DANGER))
-		remove_all_filters()
-		//Consume plasma when cancelling
-		succeed_activate()
+	var/turf/check_turf = get_step(owner, owner.dir)
+	LAZYINITLIST(targets)
+	while(check_turf && length(targets) < ZEROFORM_BEAM_RANGE)
+		targets += check_turf
+		check_turf = get_step(check_turf, owner.dir)
+	if(!LAZYLEN(targets))
+		return
+
+	var/particles_type
+	switch(owner.dir)
+		if(WEST)
+			particles_type = /particles/zero_form/west
+		if(EAST)
+			particles_type = /particles/zero_form/east
+		if(NORTH)
+			particles_type = /particles/zero_form/north
+		if(SOUTH)
+			particles_type = /particles/zero_form/south
+		else
+			particles_type = /particles/zero_form
+	particles = new(owner, particles_type)
+	beam = owner.beam(targets[length(targets)], "plasmabeam", beam_type = /obj/effect/ebeam/zeroform)
+	sound_loop.start(owner)
+	if(!do_after(owner, ZEROFORM_CHARGE_TIME, FALSE, owner, BUSY_ICON_DANGER))
+		QDEL_NULL(beam)
+		QDEL_NULL(particles)
+		targets = null
+		return fail_activate()
+	RegisterSignal(owner, list(COMSIG_MOVABLE_MOVED, COMSIG_ATOM_DIR_CHANGE), .proc/stop_beaming)
+	execute_attack()
+
+/// recursive proc for firing the actual beam
+/datum/action/xeno_action/zero_form_beam/proc/execute_attack()
+	if(!can_use_action(TRUE))
+		stop_beaming()
 		return
-	do_grav_crush(turfs)
-	remove_all_filters()
 	succeed_activate()
+	for(var/turf/target AS in targets)
+		for(var/mob/living/carbon/human/blasted in target)
```

Comment:
Not sure if its intended or not, but mechs and their pilot being completely immune to this is kinda guh.

Reply:
I can just change it later not super high priority for me